src/GenAPI/Microsoft.DotNet.GenAPI/GenAPIApp.cs

```diff
@@ -0,0 +1,185 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiSymbolExtensions;
+
+namespace Microsoft.DotNet.GenAPI
+{
+    /// <summary>
+    /// Class to standertize initilization and running of GenAPI tool.
+    ///     Shared between CLI and MSBuild tasks frontends.
+    /// </summary>
+    public static class GenAPIApp
+    {
+        public sealed class Context
+        {
+            /// <summary>
+            /// The path to one or more assemblies or directories with assemblies.
+            /// </summary>
+            public string[] Assembly { get; set; } = new string[0];
+
+            /// <summary>
+            /// If true, tries to resolve assembly reference.
+            /// </summary>
+            public bool ResolveAssemblyReferences { get; set; }
+
+            /// <summary>
+            /// Zero or more paths for resolving assembly references.
+            /// </summary>
+            public string[]? LibPath { get; set; }
+
+            /// <summary>
+            /// Output path. Default is the console. Can specify an existing directory as well and
+            /// then a file will be created for each assembly with the matching name of the assembly.
+            /// </summary>
+            public string? OutputPath { get; set; }
+
+            /// <summary>
+            /// Specify a file with an alternate header content to prepend to output.
+            /// </summary>
+            public string? HeaderFile { get; set; }
+
+            /// <summary>
+            /// Method bodies should throw PlatformNotSupportedException.
+            /// </summary>
+            public string? ExceptionMessage { get; set; }
+
+            /// <summary>
+            /// Indentation size in `IndentationChar`s. Default is 4.
+            /// </summary>
+            public int IndentationSize { get; set; } = 4;
+
+            /// <summary>
+            /// Indentation character: space, tabulation. Default is space.
+            /// </summary>
+            public char IndentationChar { get; set; } = ' ';
+
+            /// <summary>
+            /// The path to one or more attribute exclusion files with types in DocId format.
+            /// </summary>
+            public string[]? ExcludeAttributesFiles { get; set; }
+
+            /// <summary>
+            /// Include all API's not just public APIs. Default is public only.
+            /// </summary>
+            public bool IncludeVisibleOutsideOfAssembly { get; set; } = false;
+        }
+
+        /// <summary>
+        /// Initialize and run Roslyn-based GenAPI tool.
+        /// </summary>
+        public static void Run(Context context)
+        {
+            IAssemblySymbolLoader loader = new AssemblySymbolLoader(context.ResolveAssemblyReferences);
+
+            if (context.LibPath is not null)
+            {
+                loader.AddReferenceSearchPaths(context.LibPath);
```

Comment:
Assembly references can be different per assembly. Do you expect that the passed in references are a super-set or should GenAPI be able to handle different references per assembly? I.e. consider when both a net6.0 and net7.0 compiled assembly is passed in. Those would have entirely different assembly references.